docs/sections/user_guide/cli/tools/mode_template.rst

```diff
@@ -215,6 +186,46 @@ and a YAML file called ``values.yaml`` with the following contents:
      [2023-12-18T23:27:04]    DEBUG ---------------------------------------------------------------------
      [2023-12-18T23:27:04]    DEBUG Read initial values from values.yaml
 
+**NB**: The following examples are based on a ``values.yaml`` file with ``recipient: World`` removed.
+
+* It is an error to render a template without providing all needed values.
+
+  .. code-block:: text
+
+   $ uw template render --input-file template --values-file values.yaml
+   [2024-03-02T16:42:48]    ERROR Required value(s) not provided:
+   [2024-03-02T16:42:48]    ERROR   recipient
+   [2024-03-02T16:42:48]    ERROR Template could not be rendered.
+
+  But the ``--partial`` switch may be used to render as much as possible while passing expressions containing missing values through unchanged:
+
+  .. code-block:: text
+
+     $ uw template render --input-file template --values-file values.yaml --partial
+     Hello, {{ recipient }}!
+
+  Values may also be supplemented by ``key=value`` command-line arguments:
+
+  .. code-block:: text
+
+     $ uw template render --input-file template --values-file values.yaml recipient=Reader
+     Hello, Reader!
+
+  The optional ``-env`` switch allows environment variables to be used to supply values:
+
+  .. code-block:: text
+
+     $ export recipient=You
+     $ uw template render --input-file template --values-file values.yaml --env
+     Hello, You!
```

Comment:
Here's the new block about the `--env` switch. This example shows an explicitly `export`ed environment variable; the example below shows the use case where the environment variable is set on the command line.